libsmtutil/Exceptions.h

```diff
@@ -48,4 +48,25 @@ struct SMTLogicError: virtual util::Exception {};
 		"SMT assertion failed" \
 	)
 
+
+// Error to indicate that some problem occurred during an interaction with external solver.
+// This could be a problem with calling the solver or unexpected situation during the processing of solver's response.
+struct SMTSolverInteractionError: virtual util::Exception {};
+
+#if !BOOST_PP_VARIADICS_MSVC
+#define smtSolverInteractionAssert(...) BOOST_PP_OVERLOAD(smtSolverInteractionAssert_,__VA_ARGS__)(__VA_ARGS__)
```

Comment:
Please don't call it an "assert". It's a validation and that should be clear from the name. `smtSolverInteractionRequire()` would be a better name.

We already have the same problem with `assertThrow()` and `solUnimplementedAssert()` and I was planning to rename them eventually. It would be good not to introduce new things following this misleading pattern :)


 It's already a problem that some things like

Reply:
Oh, and also, for validations, I think the message should be mandatory. We should always give user the reason why things failed and making it optional makes it too easy to skip it (is see that a lot of those validations actually do not have messages).

Reply:
Thanks for the comments!
I've updated the code according to your suggestions.